x509-system/System/X509/Common.hs

```diff
@@ -0,0 +1,26 @@
+module System.X509.Common
+  ( withOpenSSLCertEnv
+  )
+where
+
+import Data.Foldable (asum)
+import Data.Maybe (catMaybes)
+import Data.Monoid (mconcat)
+import Data.X509.CertificateStore
+import System.Environment (lookupEnv)
+
+getOpenSslEnvs :: IO (Maybe String)
+getOpenSslEnvs =
+  asum
+    <$> traverse
+      lookupEnv
+      [ "SSL_CERT_FILE",
+        "SSL_CERT_DIR" 
+      ]
+
+withOpenSSLCertEnv :: IO CertificateStore -> IO CertificateStore
+withOpenSSLCertEnv defaultStore = do
+  overrideCertPaths <- getOpenSslEnvs
+  case overrideCertPaths of
+    Nothing -> defaultStore
+    Just certPath -> mconcat . catMaybes <$> mapM readCertificateStore [certPath]
```

Comment:
Would this be equivalent to the current right side of the case alternative?

```haskell
fromMaybe mempty <$> (readCertificateStore certPath)
```

It seems like the list pieces here are mainly to facilitate having a default in case `readCertificateStore` returns `Nothing`. What do you think of the readability of this option? I suggest it because it reads as more direct to my eyes, but I'd need your input as well.